src/update-feeds-loops/get-updatable-feeds.ts

```diff
@@ -19,122 +19,152 @@ export interface UpdatableBeacon {
-        const isBeaconSet = beaconValues.length > 1;
-        // Note, that the beacon set value/timestamp is computed as median, so single beacon update may not result in a beacon set update.
-        if (isBeaconSet && newDataFeedValue === dataFeedValue && newDataFeedTimestamp === dataFeedTimestamp) {
-          return false;
-        }
-        if (!isBeaconSet && newDataFeedTimestamp === dataFeedTimestamp) return false;
-
-        return logger.runWithContext(
-          {
-            dapiName: decodedDapiName,
-            dataFeedId,
-          },
-          () =>
-            checkUpdateCondition(
-              dataFeedValue,
-              dataFeedTimestamp,
-              newDataFeedValue,
-              newDataFeedTimestamp,
-              adjustedHeartbeatInterval,
-              adjustedDeviationThresholdCoefficient,
-              deviationReference
-            )
-        );
-      })
-      // Compute the updateable beacons.
-      .map(({ dataFeedInfo, aggregatedBeaconsWithData }) => ({
+        },
+        () =>
+          checkUpdateCondition(
+            dataFeedValue,
+            dataFeedTimestamp,
+            newDataFeedValue,
+            newDataFeedTimestamp,
+            adjustedHeartbeatInterval,
+            adjustedDeviationThresholdCoefficient,
+            deviationReference
+          )
+      );
+
+    if (dataFeedNeedsUpdate) {
+      acc.push({
         dataFeedInfo,
         updatableBeacons: aggregatedBeaconsWithData
           .filter(({ isUpdatable }) => isUpdatable)
           .map(({ beaconId, signedData }) => ({
             beaconId,
             signedData: signedData!,
           })),
-      }))
-  );
+        shouldUpdateBeaconSet: isBeaconSet,
+      });
+    } else if (isBeaconSet && asyncBeaconUpdatedeviationThresholdFactor) {
+      // 2.5. There is a special case when data feed is a beacon set that do not need to be updated but some of its beacon constituents do.
+      //      In this particular case, airseeker can update only these beacons and skip the beacon set update. This is enabled by setting a
+      //      value in asyncBeaconUpdatedeviationThresholdFactor on the config.
+      const updatableBeacons = aggregatedBeaconsWithData
+        .filter(
+          ({ isUpdatable, offChainValue, onChainValue }) =>
+            isUpdatable &&
+            offChainValue &&
+            checkUpdateCondition(
+              onChainValue.value,
+              onChainValue.timestamp,
+              offChainValue.value,
+              offChainValue.timestamp,
+              adjustedHeartbeatInterval,
+              adjustedDeviationThresholdCoefficient * BigInt(asyncBeaconUpdatedeviationThresholdFactor),
+              deviationReference
+            )
+        )
+        .map(({ beaconId, signedData }) => ({
+          beaconId,
+          signedData: signedData!,
+        }));
+      acc.push({
+        dataFeedInfo,
+        updatableBeacons,
```

Comment:
Isn't there a chance that `updatableBeacons` is empty array?

Reply:
Really good catch 👏🏻 I guess we should only push to the accumulator if updatableBeacons.length > 0

Reply:
https://github.com/api3dao/airseeker/pull/343/commits/b54cf23fc559cc23b4bd33c6cb5aac5170bc9536